operators/operator_sdk/osdk-new-scorecard.adoc

```diff
@@ -0,0 +1,358 @@
+[id="osdk-new-scorecard"]
+= Validating Operators using the scorecard
+include::modules/common-attributes.adoc[]
+:context: osdk-new-scorecard
+
+toc::[]
+
+:FeatureName: The enhanced Operator SDK scorecard
+include::modules/technology-preview.adoc[leveloffset=+1]
+
+Operator authors should validate that their Operator is packaged correctly and
+free of syntax errors. As an Operator author, you can use the scorecard tool in
+the Operator SDK to validate your Operator packaging and run tests.
+
+[NOTE]
+====
+The implementation of the scorecard tool supported in earlier versions of
+{product-title} 4, which was available using the `operator-sdk scorecard`
```

Comment:
We're claiming support of OSDK 0.19.4 in OCP 4.6, which doesn't have `operator-sdk screocard` available anymore (checked with @jmccormick2001 as well):

```
$ operator-sdk version
operator-sdk version: "v0.19.4", commit: "125d0dfcc71fef4f9d7e2a42b1354cb79ffdee03", kubernetes version: "v1.18.2", go version: "go1.13.15 linux/amd64"

$ operator-sdk scorecard
Error: unknown command "scorecard" for "operator-sdk"
Run 'operator-sdk --help' for usage.
FATA[0000] unknown command "scorecard" for "operator-sdk" 
```

Reply:
@adellape I tried v0.19.3 which has that. now I tried v0.19.4 which has no that. so, you are correct, and please ignore my comment.

Reply:
@adellape sorry. I build operator-sdk with the latest code (commit id ef0bc885e40e6c86c0c079bea612b4eb908f6ec5 which is latest than your version). There is still scorecard.
```console
[root@preserve-olm-env memcached-operator-scorecard]# /root/kuiwang/bin/operator-sdk version
operator-sdk version: "v0.19.4-5-gef0bc88", commit: "ef0bc885e40e6c86c0c079bea612b4eb908f6ec5", kubernetes version: "v1.18.2", go version: "go1.14 linux/amd64"
[root@preserve-olm-env memcached-operator-scorecard]# /root/kuiwang/bin/operator-sdk -h
An SDK for building operators with ease

Usage:
  operator-sdk [command]

Available Commands:
  add         Adds a controller or resource to the project
  alpha       Run an alpha subcommand
  build       Compiles code and builds artifacts
  bundle      Manage operator bundle metadata
  cleanup     Delete and clean up after a running Operator
  completion  Generators for shell completions
  generate    Invokes a specific generator
  help        Help about any command
  new         Creates a new operator application
  olm         Manage the Operator Lifecycle Manager installation in your cluster
  print-deps  Print Golang packages and versions required to run the operator
  run         Run an Operator in a variety of environments
  scorecard   Run scorecard tests
  test        Tests the operator
  version     Prints the version of operator-sdk

Flags:
  -h, --help      help for operator-sdk
      --verbose   Enable verbose logging

Use "operator-sdk [command] --help" for more information about a command.
[root@preserve-olm-env memcached-operator-scorecard]# /root/kuiwang/bin/operator-sdk scorecard -h
Runs blackbox scorecard tests on an operator

Usage:
  operator-sdk scorecard [flags]

Flags:
  -b, --bundle string       OLM bundle directory path, when specified runs bundle validation
      --config string       config file (default is '<project_dir>/.osdk-scorecard.yaml'; the config file's extension and format must be .yaml
  -h, --help                help for scorecard
      --kubeconfig string   Path to kubeconfig of custom resource created in cluster
  -L, --list                If true, only print the test names that would be run based on selector filtering
  -o, --output string       Output format for results. Valid values: text, json (default "text")
  -l, --selector string     selector (label query) to filter tests on
      --version string      scorecard version. Valid values: v1alpha2 (default "v1alpha2")

Global Flags:
      --verbose   Enable verbose logging
[root@preserve-olm-env memcached-operator-scorecard]#  /root/kuiwang/bin/operator-sdk scorecard  --kubeconfig /root/kubeconf/kubeconfig -o text
INFO[0000] Using config file: /root/kuiwang/github/operator-framework/operator-samples-for-sdk-019/ansible/memcached-operator-scorecard/.osdk-scorecard.yaml 
I1012 01:54:23.239078    6757 request.go:621] Throttling request took 1.121372108s, request: GET:https://api.kui101229.qe.devcluster.openshift.com:6443/apis/scheduling.k8s.io/v1?timeout=32s
basic:
	Spec Block Exists                   : pass
	CR: example-memcached
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"checkspectest"
...
```
so, need to change description for it. Thanks

Reply:
@kuiwang02 is correct.

v0.19.x has a dual-mode CLI, where it supports legacy projects (with a deprecation warning) and new kubebuilder-based projects.

If `operator-sdk` is executed in a "legacy" project, it reverts to the old CLI that supports the old scorecard. Otherwise it uses the new kubebuilder-based CLI.

Reply:
Thanks for helping clarify, all! Relevant code:

https://github.com/operator-framework/operator-sdk/blob/125d0dfcc71fef4f9d7e2a42b1354cb79ffdee03/cmd/operator-sdk/main.go#L39-L58